### ISSUE TITLE
Scores depending on multiple consecutive samples in time-based data

### ISSUE THREAD

**Issue by Elroch** [deeplearning4j/deeplearning4j#1274]
A quite common situation in a neural network that feeds into behaviour (eg motor actuation in a robot) is that a cost is a function of different derivatives of a target. The simplest case is where there is a single output p_t depending on timestep t and the cost is some error function:
E(p_t, p_t-1)
rather than just E(p_t)
In general p may be a vector of outputs and also E may depend on k consecutive time steps (which can be viewed as being dependence on derivatives up to the (k-1)th).  In the simplest case, the gradient of E with respect to p_t has two components coming from adjacent time steps.

In training, what would do the job is to take snippets of the time series of length L (L could be as small as k, but sometimes might be more efficient to be larger) and to do a form of minibatch gradient descent using L-k-1 examples from each snippet (those where E can be calculated).

Hope I have expressed that clearly enough. If not, please tell me!

**Comment by agibsonccc** [comment]
Come in to gitter for another explanation or ask for clarifying docs. Thanks!
